doc/mgmt/SONiC_PIM_SM_HLD.md

```diff
@@ -0,0 +1,1023 @@
+
+Configuration sequence,
+1. FRR sends update to ipmcfpmsyncd with IIF as Vlan and OIF as empty
+2. ipmcfpmsyncd populates S,G with IIF as VLAN and OIF empty in APPL_DB
+3. ipmcorchagent, upon receiving the update adds IIF VLAN to OIF list(class MRouteEntry and member list m_oifs updated with IIF vlan)<br>
+    a) calls IPMC group creation incase of L3 VLAN <br>
+    b) call L2MC group creation incase of snooping enabled on the VLAN <br>
+4. SAI expands the L2/L3 portbitmap for IIF/OIF Vlans
+
+| Source IP       | Group IP        | Incoming Interface   |Outgoing Interface(s)
+|----------------|----------------|--------------------| ---------------------
+|30.0.0.3 |        225.1.1.1|        Vlan300|              Vlan100
+| | | |    Vlan300
+|30.0.0.4 |        225.1.1.1|        Vlan300|              Vlan300
+
+#### 3.1.2.3 IPMC Aging time
+
+IPMC aging timeout is switch attribute for SAI. This is global configuration for a switch and applicable to all VRF, all AFI's.
+
+Following the sequence of configuration,
+1. FRR sends update to ipmcfpmsyncd with aging value using struct mfpm_kat_info_t.
+2. ipmcfpmsyncd populate "ipmc_aging_time" entry in  SWITCH_TABLE (APPL_DB)
+3. ipmcorchagent, receives SWITCH_TABLE (APPL_DB) update and calls set_switch_attribute with attr as SAI_SWITCH_ATTR_IPMC_AGING_TIME and value "ipmc_aging_time" <br>
+
+### 3.1.3 Syncd changes
+#### 3.1.3.1 Changes required to process IPMC entries AGE and DONTAGE data
+Following will be added in the syncd inorder to process the IPMC entry or entries AGE and DONTAGE data,
+
+1. Establish a connection with STATE_DB inorder to add or remove IPMC entries AGE and DONTAGE data.
+2. New notification handler (Notificationhandler::onIpmcEvent) API will be defined and registered with SAI to receive IPMC specific AGE and
+   DONTAGE notifications.
+   IPMC specific notification will be identified with the enum "SAI_SWITCH_NOTIFICATION_NAME_IPMC_EVENT" which will be added 
+   newly in SAI layer. 
+3. On receiving the IPMC entry/entries AGE or DONTAGE data, required validations will be done after 
+   deserializing the received data and then serialized unmodified data will be written to STATE_DB for further processing. A new API
+   will be defined to add or delete the data to/from the STATE_DB.
+
+##### 3.1.3.1.2 IPMC Entry deletion
+Whenever a IPMC entry is deleted for any reason, AGE and DONTAGE data belongs to that particular IPMC entry will be
+cleared from the STATE_DB.
+##### 3.1.3.1.3 SYNCD WARM restart
+To be updated based on the syncd warm restart feature understanding from broadcom.
+## 3.2 Datastructure Changes
+
+Struct ipmc_fpm_msg_hdr_t
+
+|    Data Type     | Field_name | Description |
+| -----------------| ---------- | ----------- |
+|  uint8_t         |  version   | Protocol version |
+|  ipmc_fpm_msg_type_t |  msg_type  | message type     |
+|  uint16_t        |  msg_len   | Length of the entire message including header length |
+
+Enum mfpm_msg_type_t
```

Comment:
Let's update the name for mfpm_msg_type_t. Also for mfpm_route_status_t down below. mfpm_kat_info_t few lines above. Also in some places, we refer MFPM client & MPFM channel, let's correct that as well.

Reply:
Updated